enterprise/views.py

```diff
@@ -294,3 +297,59 @@ def post(self, request):
             # We're handing consent only for a particular course
             return self.post_course_specific_consent(request, specific_course, consent_provided)
         return self.post_account_consent(request, consent_provided)
+
+
+class RouteLearnerData(APIView):
+    """
+    Provide an endpoint to route learner data to the appropriate channels based on Enterprise's configuration
+    """
+
+    def get(self, request, format=None):
+        """
+        It is unclear if a get request should do anything at this point.
+        """
+        return Response()
+
+    def post(self, request, format=None):
+        """
+        This endpoint is meant to be hit when certificates are generated or when audit learners in self paced courses
+        reach a passing grade. When this happens, this method will route the learners data appropriately.
+
+        For a given learner who has completed a course, find out the following:
+            if the learner is associated with an enterprise customer.
+            if the course is part of an enterprise enrollment.
+            if the enterprise customer is configured for SuccessFactors.
+
+        Continue if all the conditions above are met, otherwise abort processing.
+
+        Based on the EnterpriseCustomerPluginConfiguration for the Enterprise, determine which endpoints are
+        configured to receive learner data, and send a formatted request to each of those endpoints.
```

Comment:
We should make it clear which part of the processing chain here is happening asynchronously. Since we are creating a web API endpoint here, I was assuming that LMS would add a task to the celery queue which posts to the RouteLearnerData endpoint and all of the work documented here would happen in the same process. It might be good to throw together a quick diagram which illustrates the process flow.